packages/core/src/Site/index.js

```diff
@@ -523,6 +526,10 @@ class Site {
         : filteredPage;
     });
     this.addressablePages = Object.values(filteredPages);
+    this.addressablePagesSource.clear();
```

Comment:
this is pseudo code (there's no `clear()` function 😅)

do retest when making changes!

Reply:
Ah, right. That was a very careless mistake. Will make sure I retest every time from now on!